### PR TITLE
Add experimental "auditLog" site config setting.

### PR DESCRIPTION
**Pull request by vrto**
## Description

Add an experimental site config setting to enable audit log.

The setting is selective, the admin will specify which parts of the audit log they want enabled.

```
  "experimentalFeatures": {
    "auditLog": {
      "securityEvents": true,
      "graphQL": false,
      "gitserverAccess": false
    }
  },
```

## Test plan

Manually checked that the autocompletion in site config works.

### CODE REVIEW

**Comment by vrto** [comment]
@bobheadxi I know we already have a `log` object in `experimentalFeatures`, but I think we better keep `auditLog` top level instead of nesting it under `log` one. I like to think of this as a separate feature rather than a log setting. Do you see any potential issues with that?

**Review by rafax — APPROVED** [review]

**Comment by rafax** [comment]
> @rafax we talked about the ability to have the audit log “selective” is this aligned with what you expected?
> 
> Also, we talked that we ought to default to having security events enabled on MI; so far, I made `auditLog.securityEvents` default to `true,` but I assume we’ll need more. I’d wait until this is GA, but can you share more details on how we’ll make it happen later on?

For now, we automatically override certain siteConfig values on Cloud from a cron-like job. 
In the future, I hope we will stop supporting site-configs updates in UI, and it will become part of artefacts that are checked in into Git.

**Comment by bobheadxi** [comment]
> I know we already have a log object in experimentalFeatures, but I think we better keep auditLog top level instead of nesting it under log one. I like to think of this as a separate feature rather than a log setting. Do you see any potential issues with that?

Note that we do already have a log object at the top level (not experimentalFeatures) which is currently home to `log: { gitserver.accessLogs }`: https://sourcegraph.com/github.com/sourcegraph/sourcegraph/-/blob/schema/site.schema.json?L1164=

That said, as an experimental feature I think this is fine - once it is "promoted" to a stable feature however I think it should go under the `log` configuration instead.

Another note: I suggest that this configuration go under `log` immediately, rather than under `experimentalFeatures` - users of `experimentalFeatures` must eventually manually edit the configuration to migrate it to the new settings, and we already have plenty of non-experimentalFeatures config that are simply marked as `EXPERIMENTAL` in the help text, which I think is more than adequate to indicate the nature of the setting.

File: schema/site.schema.json [hunk 1]
```diff
@@ -534,6 +534,36 @@
...
+        "auditLog": {
+          "description": "EXPERIMENTAL: Audit Log",
+          "type": "object",
+          "additionalProperties": false,
+          "properties": {
+            "securityEvents": {
+              "description": "Capture security events as part of the audit log.",
+              "type": "boolean",
+              "default": true
+            },
```
Comment (bobheadxi) [inline schema/site.schema.json:547]: You might already know this but heads up that when accessing this config from Go code, if I remember correctly non-zero-value-defaults will _not_ be set, it must be inferred in-code

**Comment by bobheadxi** [comment]
> Does this mean it can somehow (but how?) be marked as 'experimental' even though we'll nest it under log straight away?

No, but it is made very visible in the docstring on the config, where presumably anyone who doesn't know what the setting does will see:

image

> Also, isn't a top-level auditLog a little cleaner than nesting it under a log? I know it kinda looks like "logging stuff," but I imagined that log properties would be used for configuring the loggers, while auditLog properties are selecting which parts make it to the audit log. I'm relatively easy, but I wanted to point the difference out.

That's one way of thinking about it, but in my head at the end it is a type of logging, and as a site admin I would hope that all logging-related config (what to log, how to log it, etc) be in one place to make my life marginally easier 😅

**Comment by vrto** [comment]
@bobheadxi 

re-experimental: sounds good, will refactor
re-nesting under log: ok, I see your point :) I'll think about it a little more and push a change some time tomorrow.

**Review by dcomas — APPROVED** [review]

**Pull request merged**